querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java

```diff
@@ -895,13 +895,7 @@ protected void visitOperation(Class<?> type, Operator operator, List<? extends E
             }
         }
 
-        if (operator == Ops.LIST && args.get(0) instanceof SubQueryExpression) {
-            boolean oldUnion = inUnion;
-            inUnion = true;
-            super.visitOperation(type, SQLOps.UNION, args);
-            inUnion = oldUnion;
-
-        } else if (operator == SQLOps.UNION || operator == SQLOps.UNION_ALL) {
+        if (operator == SQLOps.UNION || operator == SQLOps.UNION_ALL) {
```

Comment:
The code I've removed looks wrong to me, and breaks coalesce for subqueries. Was it doing anything useful?